proto/process/agent.proto

```diff
@@ -463,6 +463,52 @@ message CollectorStatus {
 	int32 interval = 2;
 }
 
+enum Language {
+	LANGUAGE_UNKNOWN = 0;
+	LANGUAGE_JAVA = 1;
+	LANGUAGE_NODEJS = 2;
+	LANGUAGE_PYTHON = 3;
+	LANGUAGE_RUBY = 4;
+	LANGUAGE_DOTNET = 5;
+	LANGUAGE_GO = 6;
+	LANGUAGE_CPLUSPLUS = 7;
+	LANGUAGE_PHP = 8;
+}
+
+enum ServiceNameSource {
+	SERVICE_NAME_SOURCE_UNKNOWN = 0;
+	SERVICE_NAME_SOURCE_DD_SERVICE = 1;
+	SERVICE_NAME_SOURCE_COMMAND_LINE = 2;
+	SERVICE_NAME_SOURCE_LARAVEL = 3;
+	SERVICE_NAME_SOURCE_PYTHON = 4;
+	SERVICE_NAME_SOURCE_NODEJS = 5;
+	SERVICE_NAME_SOURCE_GUNICORN = 6;
+	SERVICE_NAME_SOURCE_RAILS = 7;
+	SERVICE_NAME_SOURCE_SPRING = 8;
+	SERVICE_NAME_SOURCE_JBOSS = 9;
+	SERVICE_NAME_SOURCE_TOMCAT = 10;
+	SERVICE_NAME_SOURCE_WEBLOGIC = 11;
+	SERVICE_NAME_SOURCE_WEBSPHERE = 12;
+}
+
+// TracerMetadata as defined in
+// https://github.com/DataDog/libdatadog/blob/99056cf717cfe9/ddcommon/src/tracer_metadata.rs#L7-L29
+message TracerMetadata {
```

Comment:
Do you have an idea of how large these metadata will be and how they will be used? It also seems like some fields are duplicates, e.g. `hostname` or `service_name`, could they be derived from the top-level fields or is it actually possible for them to differ between tracers on the same process?